asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go

```diff
@@ -475,6 +475,79 @@ func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface)
 
 }
 
+// DeleteAsset can be used by the owner of the asset to delete the asset
+func (s *SmartContract) PurgeAsset(ctx contractapi.TransactionContextInterface) error {
+
+	transientMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return fmt.Errorf("Error getting transient: %v", err)
+	}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientDeleteJSON, ok := transientMap["asset_purge"]
+	if !ok {
+		return fmt.Errorf("asset to purge not found in the transient map")
+	}
+
+	type assetPurge struct {
+		ID string `json:"assetID"`
+	}
+
+	var assetPurgeInput assetPurge
+	err = json.Unmarshal(transientDeleteJSON, &assetPurgeInput)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %v", err)
+	}
+
+	if len(assetPurgeInput.ID) == 0 {
+		return fmt.Errorf("assetID field must be a non-empty string")
+	}
+
+	// Verify that the client is submitting request to peer in their organization
+	err = verifyClientOrgMatchesPeerOrg(ctx)
+	if err != nil {
+		return fmt.Errorf("DeleteAsset cannot be performed: Error %v", err)
+	}
+
+	log.Printf("Deleting Asset: %v", assetPurgeInput.ID)
+	valAsbytes, err := ctx.GetStub().GetPrivateData(assetCollection, assetPurgeInput.ID) //get the asset from chaincode state
+	if err != nil {
+		return fmt.Errorf("failed to read asset: %v", err)
+	}
+	if valAsbytes == nil {
+		return fmt.Errorf("asset not found: %v", assetPurgeInput.ID)
+	}
+
+	ownerCollection, err := getCollectionName(ctx) // Get owners collection
+	if err != nil {
+		return fmt.Errorf("failed to infer private collection name for the org: %v", err)
+	}
+
+	//check the asset is in the caller org's private collection
+	valAsbytes, err = ctx.GetStub().GetPrivateData(ownerCollection, assetPurgeInput.ID)
+	if err != nil {
+		return fmt.Errorf("failed to read asset from owner's Collection: %v", err)
+	}
+	if valAsbytes == nil {
+		return fmt.Errorf("asset not found in owner's private Collection %v: %v", ownerCollection, assetPurgeInput.ID)
+	}
+
+	// delete the asset from state
+	err = ctx.GetStub().DelPrivateData(assetCollection, assetPurgeInput.ID)
```

Comment:
I assume any reference to DelPrivateData() should simply be changed to PurgePrivateData()